internal/controller/log.go

```diff
@@ -48,15 +51,25 @@ func (d *DeployLog) LogDeploy(timeStamp time.Time, podName string, podIP string,
 		Reason:           reason,
 	}
 
-	d.Entries = append(d.Entries, newEntry)
+	if len(d.entries) >= d.maxEntries {
+		// Pull elements off the front of the slice to make sure that the newly appended record is under the max record value.
+		d.entries = d.entries[(len(d.entries)-d.maxEntries)+1:]
+	}
+
+	d.entries = append(d.entries, newEntry)
 }
 
 // GetLog returns a json representation of the entries list.
 func (d *DeployLog) GetLog() []byte {
-	data, err := json.Marshal(d.Entries)
+	data, err := json.Marshal(d.entries)
 	if err != nil {
 		log.Error("Could not marshal deploylog entries")
 	}
 
 	return data
 }
+
+// GetLogLength returns the number of records in the entries slice.
+func (d *DeployLog) GetLogLength() int {
```

Comment:
This would probably mean a `deployLog.GetLogLenth()`, WDYT of this instead ?
```suggestion
func (d *DeployLog) Len() int {
```

Reply:
Also where it is called except from tests ?

Reply:
👍